### ISSUE TITLE
addClass is not working for the second time

### ISSUE THREAD

**Issue by boldcolin** [luruke/barba.js#244]
Everything works fine after the init. But going back to homepage and then back to a case page the 
 
$slides.addClass('transition'); 

is not adding a class to the div.

I feel that it has something todo with not being able to find the element.. Any idea?

    document.addEventListener("DOMContentLoaded", function() {

    // assign some variables
    var lastElementClicked;

    var isAnimating = false;

    var $body = document.querySelector('body');
    var $html = document.querySelector('html');

    var $slides = $('.slides');

    // options
    Barba.Pjax.Dom.wrapperId = 'content-wrapper';
    Barba.Pjax.Dom.containerClass = 'content-container';

    //Barba.Pjax.cacheEnabled = false;

    // ---------------- //
    // VIEWS
    // ---------------- //
    var homeView = Barba.BaseView.extend({
        namespace: 'home',

        onEnter: function() {

            console.log('home - onEnter');

            // The new Container is ready and attached to the DOM.
            HOME.init();

        },
        onEnterCompleted: function() {
            console.log('home - onEnterCompleted');
        },

        onLeave: function() {

            console.log('home - onLeave');

        },

        onLeaveCompleted: function() {

            console.log('home > onLeaveCompleted');

        },

    });

    var caseView = Barba.BaseView.extend({
        namespace: 'case',
        onEnterCompleted: function() {

            console.log('case > onEnterCompleted');

        },
        onLeave: function() {

            console.log('case > onLeave');

        },
        onLeaveCompleted: function() {

            console.log('case > onLeaveCompleted');
        },
    });
    //
    // // Don't forget to init the view!
     homeView.init();
     caseView.init();

    Barba.Pjax.init();

    // listen to the event on click
    // can now reference lastElementClicked to scroll to where it's been clicked
    Barba.Dispatcher.on('linkClicked', function(el) {
        lastElementClicked = el;
    });

    // -------------------- //
    // TRANSITION FUNCTIONS
    // -------------------- //
    var revealProject = Barba.BaseTransition.extend({

        start: function() {

            isAnimating = true;
            console.log('revealProject > start');

            // set up functions asynchronously
            Promise
                .all([this.newContainerLoading, this.removeMask()])
                .then(this.showNewPage.bind(this));

        },

        // first transition function
        removeMask: function() {

            var deferred = Barba.Utils.deferred();

            var $slides = $('.slides');

            $slides.addClass('transition');
            console.log('REMOVE MASK!');

            deferred.resolve();

            return deferred.promise;

        },

        // transition to new page / object
        showNewPage: function() {

            // assign objects that are transitioning
            var _this = this;

            isAnimating = false;
            _this.done();
        }}
    );

    // -------------------- //
    // SET TRANSITIONS
    // -------------------- //
    Barba.Pjax.getTransition = function() {

        var transitionPage = revealProject;

        return transitionPage;
    };
    });

**Comment by mattiaband** [comment]
I think that I have the same problem.
I am not able to trigger the click on the second, third ecc page, the click works just if in the first page.

`
$(".xxxxx").on('click',function(){        
   console.log('clicked');
})
`

@boldcolin did you find a solution?

**Comment by benjaminpreiss** [comment]
Did anyone find a solution to this? I am encountering the same issue...

**Comment by benjaminpreiss** [comment]
Ah, in my case I ran jquerys `toggleClass` on `barba.hooks.after()`. On every page transition a click event with `toggleClass` was added. Moving to the `once` hook solved the problem.
